apps/linear_algebra/src/blas_l3_generators.cpp

```diff
@@ -30,157 +25,100 @@ class GEMMGenerator :
     Param<T>   b_ = {"b", 1.0};
     ImageParam C_ = {type_of<T>(), 2, "C"};
 
-    void SetupTarget() {
-        if (!assertions_enabled_) {
-            target.set(get_target()
-                       .with_feature(Target::NoAsserts)
-                       .with_feature(Target::NoBoundsQuery));
-        }
-
-        if (use_fma_) {
-            target.set(get_target().with_feature(Target::FMA));
-        }
-    }
-
-    Func transpose(ImageParam im) {
-        Func transpose_tmp("transpose_tmp"), im_t("im_t");
-        Var i("i"), j("j"), ii("ii"), ji("ji"),
-            ti("ti"), tj("tj"), t("t");
-
-        transpose_tmp(i, j) = im(j, i);
-        im_t(i, j) = transpose_tmp(i, j);
-
-        Expr rows = im.width(), cols = im.height();
-
-        im_t.compute_root()
-            .specialize(rows >= 4 && cols >= 4)
-            .tile(i, j, ii, ji, 4, 4).vectorize(ii).unroll(ji)
-            .specialize(rows >= 128 && cols >= 128)
-            .tile(i, j, ti, tj, i, j, 16, 16)
-            .fuse(ti, tj, t).parallel(t);
-
-        transpose_tmp.compute_at(im_t, i)
-            .specialize(rows >= 4 && cols >= 4).vectorize(j).unroll(i);
-
-        return im_t;
-    }
+    Var i, j, ii, ji, jii, io, jo, ti, tj, t;
 
     Func build() {
-        SetupTarget();
+        const Expr num_rows = (A_.width()/32)*32;
+        const Expr num_cols = (B_.height()/32)*32;
+        const Expr sum_size = (A_.height()/32)*32;
```

Comment:
This was mostly do I didn't have to write a schedule that accounted for tail cases, etc. A minimum size of 32 is actually overkill for doubles, because they use smaller tiles (because natural_vector_size is smaller). dgemm performance is sane with this PR (~half the gflops of sgemm, as expected).